### PR TITLE
test: add json tests for empty array and null

### PR DESCRIPTION
**Pull request by lotyp**

### CODE REVIEW

File: tests/ORM/Functional/Driver/Common/Typecast/JsonTest.php [hunk 1]
```diff
@@ -153,6 +153,72 @@ public function testStore(): void
...
+    {
+        $e = new User();
+        $e->email = 'test@email.com';
+        $e->settings = [];
+        $e->settingsNullable = [];
+
+        $this->captureWriteQueries();
+        $em = new EntityManager($this->orm);
+        $em->persist($e);
+        $em->run();
```
Comment (roxblnfk) [inline tests/ORM/Functional/Driver/Common/Typecast/JsonTest.php]:
```suggestion
        $this-save($e);
```

File: tests/ORM/Functional/Driver/Common/Typecast/JsonTest.php [hunk 2]
```diff
@@ -153,6 +153,72 @@ public function testStore(): void
...
+        $this->assertNumWrites(1);
+
+        $this->captureWriteQueries();
+        $em = new EntityManager($this->orm);
+        $em->persist($e);
+        $em->run();
```
Comment (roxblnfk) [inline tests/ORM/Functional/Driver/Common/Typecast/JsonTest.php]:
```suggestion
        $this-save($e)
```

File: tests/ORM/Functional/Driver/Common/Typecast/JsonTest.php [hunk 3]
```diff
@@ -153,6 +153,72 @@ public function testStore(): void
...
+        $this->assertNumWrites(0);
+
+        $this->assertEquals(3, $e->id);
+
+        $this->assertTrue($this->orm->getHeap()->has($e));
+        $this->assertSame(Node::MANAGED, $this->orm->getHeap()->get($e)->getStatus());
+
+        $this->orm = $this->orm->with(heap: new Heap());
```
Comment (roxblnfk) [inline tests/ORM/Functional/Driver/Common/Typecast/JsonTest.php]:
```suggestion
        $this->orm = $this->orm->getHeap()->clear();
```

File: tests/ORM/Functional/Driver/Common/Typecast/JsonTest.php [hunk 4]
```diff
@@ -153,6 +153,72 @@ public function testStore(): void
...
+        $this->captureWriteQueries();
+        $em = new EntityManager($this->orm);
+        $em->persist($e);
+        $em->run();
+        $this->assertNumWrites(1);
+
+        $this->captureWriteQueries();
+        $em = new EntityManager($this->orm);
+        $em->persist($e);
+        $em->run();
```
Comment (roxblnfk) [inline tests/ORM/Functional/Driver/Common/Typecast/JsonTest.php]:
```suggestion
        $this-save($e)
```

File: tests/ORM/Functional/Driver/Common/Typecast/JsonTest.php [hunk 5]
```diff
@@ -153,6 +153,72 @@ public function testStore(): void
...
+    {
+        $e = new User();
+        $e->email = 'test@email.com';
+        $e->settings = [];
+        $e->settingsNullable = null;
+
+        $this->captureWriteQueries();
+        $em = new EntityManager($this->orm);
+        $em->persist($e);
+        $em->run();
```
Comment (roxblnfk) [inline tests/ORM/Functional/Driver/Common/Typecast/JsonTest.php]:
```suggestion
        $this-save($e)
```

**Review by roxblnfk — APPROVED** [review]
